content/chainguard/chainguard-enforce/authentication/identity-examples/enforce-aws-identity.md

```diff
@@ -0,0 +1,359 @@
+
+The next section will output the new identity's `id` value. This is a unique value that represents the identity itself.
+
+```
+output "aws-identity" {
+  value = chainguard_identity.aws.id
+}
+```
+
+The section after that looks up the `viewer` role.
+
+```
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+```
+
+The final section grants this role to the identity on the `example-group`.
+
+```
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.aws.id
+  group	= chainguard_group.user-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+```
+
+After defining these resources, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+## Using the identity with AWS Lambda
+
+In this section, you'll create an AWS Lambda function that will assume the identity you created in the previous section. This function will then use the identity to interact with Chainguard resources.
+
+First, define the following resources in a `lambda.tf` file. The file will consist of three sections, which we'll go over one by one.
+
+This section defines an IAM policy document that will allow the AWS Lambda service to assume the role you'll create in the next section:
+
+```hcl
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+```
+
+This section defines the role that the AWS Lambda service will assume:
+
+```hcl
+resource "aws_iam_role" "iam_for_lambda" {
+  name               = "iam_for_lambda"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+```
+
+This section defines a AWS lambda function implemented in Node.js that will assume the identity you created in the previous section:
+
+```hcl
+resource "aws_lambda_function" "test_lambda" {
+  filename      = "lambda_function_payload.zip"
+  function_name = "lambda_function_name"
+  role          = aws_iam_role.iam_for_lambda.arn
+  handler       = "index.test"
+
+  source_code_hash = data.archive_file.lambda.output_base64sha256
+
+  runtime = "nodejs18.x"
+
+  environment {
+    variables = {
+      foo = "bar"
+    }
+  }
+}
+```
+
+When this function is invoked, it will assume the AWS role you created in the previous section, and be able to present credentials as that AWS role that will allow it to assume the Chainguard identity you created in the previous section, to view and manage Chainguard resources.
+
+## Creating Your Resources
+
+First, run `terraform init` to initialize Terraform's working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
```

Comment:
```suggestion
terraform plan
```
Again, i could not get past this command successfully. I'll still review the rest of the content but I'll need to pass a tech test before I can approve it.